### PR TITLE
Add QuartzNet model

### PR DESCRIPTION
**Pull request by jkamelin**

### CODE REVIEW

File: models/public/quartznet/accuracy-check.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    launchers:
+      - framework: dlsdk
+        allow_reshape_input: True
+        adapter:
+          type: ctc_greedy_decoder
+          alphabet: [' ', a, b, c, d, e, f, g, h, i, j, k, l, m, n, o, p, q, r, s, t, u, v, w, x, y, z, '''']
+        inputs:
+          - name: 'audio_signal'
+            type: INPUT
+            layout: NHWC
```
Comment (eaidova) [inline models/public/quartznet/accuracy-check.yml:12]: are you sure in layout?
I believe input is 3-dim only, not 4

File: models/public/quartznet/quartznet.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+#### Original model
+
+Per-frame probabilities (after LogSoftmax) for every symbol in the alphabet, name - `output`,  shape - `1, 64, 29`, output data format is `B, N, C` where:
+
+- B - batch size
+- N - number of audio frames
+- C - alphabet size, including the CTC blank symbol
+
+The per-frame probabilities are to be decoded with a CTC decoder.
+The alphabet is: 0 = space, 1...26 = "a" to "z", 27 = apostrophe, 28 = CTC blank symbol.
```
Comment (eaidova) [inline models/public/quartznet/quartznet.md:51]: maybe we can provide the file with alphabet?

File: models/public/quartznet/quartznet.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+Mozilla Common Voice, WSJ, Fisher, Switchboard, and NSC Singapore English.
+For details see [repository](https://github.com/NVIDIA/NeMo), [paper](https://arxiv.org/pdf/1910.10261.pdf).
+
+## Accuracy
+
+| Metric                       | Value |
+| ---------------------------- | ----- |
+| WER @ Librispeech test-clean | 3.86% |
+
+## Specification
```
Comment (eaidova) [inline models/public/quartznet/quartznet.md:17]: as I know, we usually define specification, then accuracy.
Could you please swap them?

**Comment by eaidova** [comment]
@Wovchena FYI, model will be added as joint encoder+decoder, possibly you need update https://github.com/openvinotoolkit/open_model_zoo/pull/2054

File: models/public/quartznet/model.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import torch
+import yaml
```
Comment (IRDonch) [inline models/public/quartznet/model.py:17]: I think depending on PyYAML here is a bit fragile. It's not actually declared as a conversion dependency; the only reason it's available in the conversion environment is because it's a dependency of yacs, which is a dependency of another model (`regnetx-3.2gf`). So if we remove that model, we might end up removing PyYAML as well.

I would suggest to instead add the ruamel.yaml wheel to the downloaded files (it's pure Python) and use that here. It'll also let you drop a couple of regex hacks.

File: models/public/quartznet/quartznet.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
+# quartznet
+
+## Use Case and High-Level Description
+
+QuartzNet model is a pair of encoder and decoder. QuartzNet’s design is based on the Jasper architecture,
```
Comment (IRDonch) [inline models/public/quartznet/quartznet.md:5]: > QuartzNet model is a pair of encoder and decoder.

I think that's rather misleading, given that we merged them.

Furthermore, this is missing the most important information - what does this model actually _do_?

File: models/public/quartznet/quartznet.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+### Input
+
+#### Original model
+
+Audio signal, name - `audio_signal`,  shape - `1, 64, 128`, format is `B, N, C` where:
+
+- `B` - batch size
+- `N` - number of audio frames
+- `C` - clip duration
+
```
Comment (IRDonch) [inline models/public/quartznet/quartznet.md:35]: This is lacking some important information about the expected data: what is the format of the samples? (-1..1? 0..65535?) What is the sample rate?

**Review by IRDonch — CHANGES_REQUESTED** [review]

File: models/public/quartznet/accuracy-check.yml (hunk 1 above)
Comment (jkamelin) [reply models/public/quartznet/accuracy-check.yml:12]: The input is 3-dim but [accuracy_checker/launcher/dlsdk_launcher.py#L846](https://github.com/openvinotoolkit/open_model_zoo/blob/develop/tools/accuracy_checker/accuracy_checker/launcher/dlsdk_launcher.py#L846) returns 4-dim `data_shape`. Do you have any suggestions to avoid such behavior?

Comment (eaidova) [reply models/public/quartznet/accuracy-check.yml:12]: ok, then I will take it in account, I do not think that we should resolve this issue as part of this request

**Comment by eaidova** [comment]
@jkamelin please do not forget to update model name

File: models/public/quartznet/quartznet.md (hunk 3 above)
Comment (jkamelin) [reply models/public/quartznet/quartznet.md]: done

File: models/public/quartznet/quartznet.md (hunk 2 above)
Comment (jkamelin) [reply models/public/quartznet/quartznet.md]: We have one in [speech_recognition_demo/python/default_alphabet_example.conf](https://github.com/openvinotoolkit/open_model_zoo/blob/develop/demos/speech_recognition_demo/python/default_alphabet_example.conf). Added link.

File: models/public/quartznet/quartznet.md (hunk 5 above)
Comment (jkamelin) [reply models/public/quartznet/quartznet.md]: Description updated

File: models/public/quartznet/model.py (hunk 4 above)
Comment (jkamelin) [reply models/public/quartznet/model.py]: Done

File: models/public/quartznet/model.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+        self.encoder._prepare_for_deployment()
+        self.decoder._prepare_for_deployment()
+
+        type(self.encoder).__call__ = torch.nn.Module.__call__
+        type(self.decoder).__call__ = torch.nn.Module.__call__
+
+    def eval(self):
+        self.encoder.eval()
+        self.decoder.eval()
```
[the remark this answers is not in the document]
Comment (jkamelin) [reply models/public/quartznet/model.py]: It does, removed `eval`

File: models/public/quartznet/model.py [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+        decoder_params = model_args['init_params']['decoder_params']['init_params']
+        self.decoder = JasperDecoderForCTC(**decoder_params)
+        self.decoder.load_state_dict(torch.load(decoder_weights, map_location='cpu'))
+
+        self.encoder._prepare_for_deployment()
+        self.decoder._prepare_for_deployment()
+
+        type(self.encoder).__call__ = torch.nn.Module.__call__
+        type(self.decoder).__call__ = torch.nn.Module.__call__
```
[the remark this answers is not in the document]
Comment (jkamelin) [reply models/public/quartznet/model.py]: Done

File: models/public/quartznet/quartznet.md (hunk 6 above)
Comment (jkamelin) [reply models/public/quartznet/quartznet.md]: Updated input information

**Review by IRDonch — APPROVED** [review]

**Comment by eaidova** [comment]
Jenkins please retry a build

File: models/public/quartznet-15x5-en/quartznet-15x5-en.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
+# quartznet-15x5-en
+
+## Use Case and High-Level Description
+
+QuartzNet model performs automatic speech recognition. QuartzNet’s design is based on the Jasper architecture,
+which is a convolutional model trained with Connectionist Temporal Classification (CTC) loss.
+The model was trained in NeMo on multiple datasets: LibriSpeech,
+Mozilla Common Voice, WSJ, Fisher, Switchboard, and NSC Singapore English.
+For details see [repository](https://github.com/NVIDIA/NeMo), [paper](https://arxiv.org/pdf/1910.10261.pdf).
+
```
Comment (eaidova) [inline models/public/quartznet-15x5-en/quartznet-15x5-en.md:10]: I think if you add 15x5 in model name, you need to provide description what does it is mean (at least mention this architecture detail)

Comment (jkamelin) [reply models/public/quartznet-15x5-en/quartznet-15x5-en.md:10]: Done

**Review by eaidova — APPROVED** [review]

**Review by vladimir-dudnik — APPROVED** [review]
